File: JS/WaterBottles.js
/**
 * @version JAVASCRIPT ECMAScript 6
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */

/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
function numWaterBottles(numBottles, numExchange) {
    if (numBottles < numExchange) {
        return numBottles;
    }

    let quotient = Math.floor(numBottles / numExchange);
    let reminder = numBottles % numExchange;
    let ans = numBottles + quotient;

    while (quotient + reminder >= numExchange) {
        let total = quotient + reminder;
        quotient = Math.floor(total / numExchange);
        reminder = total % numExchange;
        ans += quotient;
    }

    return ans;
};
export { numWaterBottles }

File: JAVA/WaterBottles.java
/**
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package JAVA;

public class WaterBottles {

    public static void main(String[] args) {
        System.out.printf("Resultado: %d\n", numWaterBottles( 2, 4 ));
    }

    public static int numWaterBottles(int numBottles, int numExchange) {
        if (numBottles < numExchange) {
            return numBottles;
        }

        int quotient = numBottles / numExchange;
        int reminder = numBottles % numExchange;
        int ans = numBottles + quotient;

        while (quotient + reminder >= numExchange) {
            int total = quotient + reminder;
            quotient = total / numExchange;
            reminder = total % numExchange;
            ans += quotient;
        }

        return ans;
    }

}
